relay-metrics/src/aggregation.rs

```diff
@@ -1287,7 +1339,7 @@ impl Aggregator {
 
                 let flush_at = self.config.get_flush_time(timestamp, project_key);
                 let bucket = value.into();
-                added_cost = bucket.cost();
+                added_cost = entry.key().cost() + bucket.cost();
```

Comment:
While at it, it would be good to deal with the `Occupied` branch too. See https://github.com/getsentry/relay/pull/1287#discussion_r888926787